### ISSUE TITLE
[a11y] Onboarding's Pick a Theme radio buttons are missing programmatic selected state

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#27734]
## Steps to reproduce
1. Have VoiceOver running
1. Open Firefox for the first time
2. On the onboarding screens, locate a "Pick a Theme" screen (4 of 5 in the onboarding flow)
3. Locate "System Auto", "Light", and "Dark" radio buttons and observe the VO announcement

### Expected behavior
The visually selected theme would be announced as such by VO, i.e. "System Auto, button, selected"

### Actual behavior
The Theme picker radio buttons are not announced as "selected" even when they visually are. Users of VoiceOver would have no way to learn which one is, in fact, selected, if any..

### User feedback:
> When I select a button or link, nothing happens
> When I am on the select theme screen and I select a theme option, voiceover gives me no indication that the theme I clicked on was actually selected. This means that I cannot know whether the correct theme was selected with voiceover. This could make it difficult for screen reader users to choose the theme that they want.

### Device & build information
Screenshot captured from a user feedback received on July 2, 2025, thus details on the build and iOS version are not known
* Device: ? iPhone
* Build version: (current release)
* First seen version: ?
### Notes
Attachments: 

<img width="467" alt="System Auto theme is highlighted with VO and is marked as checked visually" src="https://github.com/user-attachments/assets/358d2adf-7ede-4681-a645-849d6d6412cf" />

Follow [the Magenta11y Native App > Radio Buttons guide](https://www.magentaa11y.com/#/native-criteria/controls/radio-button?tab=3) for developer notes and testing example and checklist for radio buttons accessibility

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-12737)

**Comment by arq0017** [comment]
Hi, can you assign this issue to me ? Also I don't have access to jira Task.
